### PR TITLE
Fix a panic triggered when handling a request that yields a nil response.

### PR DESCRIPTION
**Pull request by ncabatoff**
Improve integ test to actually make changes while in recovery mode and
verify they're still there after coming back in regular mode.

### CODE REVIEW

File: vault/generate_root.go [hunk 1]
```diff
@@ -145,6 +146,9 @@ func (c *Core) GenerateRootInit(otp, pgpKey string, strategy GenerateRootStrateg
 	if c.Sealed() && !c.recoveryMode {
 		return consts.ErrSealed
 	}
+	if !c.Sealed() && c.recoveryMode {
+		return errors.New("attempt to generate recovery token when already unsealed")
```
Comment (vishalnayak) [inline vault/generate_root.go]: Can we call it `recovery operation token`?

Comment (ncabatoff) [reply vault/generate_root.go]: I liked that idea when we talked about it previously, but if the command line option and the api path say `recovery-token` I'm not sure we gain clarity by calling it a "recovery operation token" in the logs/errors.

Comment (vishalnayak) [reply vault/generate_root.go]: I don't think we can rely on core being unsealed. We may have to do `!c.barrier.Sealed()` check here.

Comment (ncabatoff) [reply vault/generate_root.go]: Mind you I guess there's precedent with dr-token.  Ok, will fix.

File: vault/generate_root.go [hunk 2]
```diff
@@ -221,6 +227,9 @@ func (c *Core) GenerateRootUpdate(ctx context.Context, key []byte, nonce string,
 	if c.Sealed() && !c.recoveryMode {
 		return nil, consts.ErrSealed
 	}
+	if !c.Sealed() && c.recoveryMode {
```
Comment (vishalnayak) [inline vault/generate_root.go]: Same here. We may have to use `!c.barrier.Sealed()`

Comment (ncabatoff) [reply vault/generate_root.go]: Not following, what's the same here?

Comment (vishalnayak) [reply vault/generate_root.go]: In recovery mode, we only care about barrier being unsealed. Core will likely remain sealed throughout.

Comment (ncabatoff) [reply vault/generate_root.go]: Ah, didn't see your other comment in that thread.  Yeah, I was setting `c.sealed=0` but I like your solution better.

**Pull request merged**